client/state/data-layer/wpcom/batch.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+
+const FLUSH_TIMEOUT = 50;
+
+const createQueue = () => {
+	let requests = [];
+
+	const flush = () => {
+		if ( ! requests.length ) {
+			return;
+		}
+		const requestsToTrigger = requests;
+		requests = [];
```

Comment:
this is a big strange to see the renaming and resetting of state. I wonder if there's a way that doesn't require this sleight-of-hand